src/main/kotlin/io/github/mojira/arisa/modules/CommandModule.kt

```diff
@@ -34,42 +38,86 @@ class CommandModule(
-            }
-            results.forEach { (comment, result) ->
-                if (result.isLeft()) {
-                    comment.update("[~arisabot]: (x) ${(result as Either.Left).a.message}.\n----\n${comment.body}")
-                } else {
-                    comment.update("[~arisabot]: (/) ${(result as Either.Right).b}.\n----\n${comment.body}")
+            val commands = staffComments
+                .map { comment ->
+                    comment to extractCommands(issue, comment)
                 }
-            }
+                .filter { it.second.isNotEmpty() }
+                .onEach { invocation ->
+                    val commandResults = invocation.second
+                        .map { it.source.line to executeCommand(it) }
+                        .toMap()
+                    editInvocationComment(invocation.first, commandResults)
+                }
+            assertNotEmpty(commands).bind()
+
             ModuleResponse.right().bind()
         }
     }
 
-    @Suppress("SpreadOperator")
-    private fun executeCommand(comment: Comment, issue: Issue): Either<Throwable, Int> {
-        val source = CommandSource(issue, comment)
+    private fun executeCommand(command: Command): CommandResult {
         return runBlocking {
             Either.catch {
-                commandDispatcher.execute(comment.body, source)
+                commandDispatcher.execute(command.command, command.source)
             }
         }
     }
 
     private fun isUpdatedAfterLastRun(lastRun: Instant, comment: Comment) = comment.updated.isAfter(lastRun)
 
-    private fun isProbablyACommand(comment: Comment) =
-        !comment.body.isNullOrBlank() &&
-                comment.body.startsWith("${prefix}_") &&
-                (comment.body.count { it.isWhitespace() } > 0)
+    /**
+     * Extracts all commands from a comment.
+     */
+    private fun extractCommands(issue: Issue, comment: Comment) =
+        comment.body?.lines().orEmpty()
+            .mapIndexed { lineNr, line -> lineNr to line.trim() }
+            .filter { (_, line) -> line.startsWith("${prefix}_") }
+            .filter { (_, line) -> line.any { char -> char.isWhitespace() } }
+            .map { (lineNr, line) ->
+                Command(line, CommandSource(issue, comment, lineNr))
+            }
+
+    /**
+     * Edits the given comment (that was used to invoke some commands) with the given results.
+     *
+     * @param results Command results mapped to the line number in the comment
+     */
+    private fun editInvocationComment(comment: Comment, results: Map<Int, CommandResult>) {
+        val newBody = comment.body?.lines().orEmpty()
+            .mapIndexed { lineNr, line ->
+                when (val result = results[lineNr]) {
+                    null -> line
+                    else -> "\n${ getCommandFeedback(line, result) }\n"
+                }
+            }
+            .joinToString("\n")
+            .replace(Regex("""^\s*|\s*$"""), "")
```

Comment:
Uhm well actually kotlin has a handy method for this called `.trim()`... I feel stupid now

Yeah agreed, that part could use some commenting.

Reply:
But using `trim()` here changes behavior: Previously it would trim leading whitespaces for every line; now it only trims it for the comment as a whole; though maybe this is acceptable?

Moving `trim()` before `joinToString("\n")` would work, but it trims all chars <= U+0020 (I think), so it would remove leading and trailing line terminators added by the string containing `getCommandFeedback` above.
You would have to use `trim(' ')` instead then.

Reply:
I'd consider this to be fine; the intention of this was just to get rid of extra `\n`s before and after the comment that were added here https://github.com/mojira/arisa-kt/blob/d0fa2f9e0eec1e888b9753d4f97bbf9d3982fcc6/src/main/kotlin/io/github/mojira/arisa/modules/CommandModule.kt#L92

So that instead of a comment looking like this

```
<empty line>
(/) SOME_COMMAND argument
→ {{command feedback}} -- arisa
<empty line>
```
it'll just remove the extra empty lines.

Arguably the behaviour before was incorrect.